src/braket/experimental/autoqasm/api.py

```diff
@@ -487,8 +541,139 @@ def _wrap_for_oqpy_gate(
                 f'Parameter "{param.name}" for gate "{f.__name__}" '
                 "must have a type hint of either aq.Qubit or float."
             )
+    return gate_args
 
-    def _func(*args: Any) -> None:
-        aq_transpiler.converted_call(f, *args, kwargs={}, options=options)
 
-    return _func, gate_args
+def _convert_calibration(
+    f: Callable,
+    options: converter.ConversionOptions,
+    args: List[Any],
+    kwargs: Dict[str, Any],
+    gate_function: Callable,
+    **decorator_kwargs,
+) -> GateCalibration:
+    """Convert the initial callable `f` into a GateCalibration object that will be added to
+    the main program as defcal.
+
+    Args:
+        f (Callable): The function to be converted.
+        options (converter.ConversionOptions): Converter options.
+        args (List[Any]): Arguments passed to the program when called.
+        kwargs (Dict[str, Any]): Keyword arguments passed to the program when called.
+        gate_function (Callable): The gate function which calibration is being defined.
+        decorator_kwargs: Keyword arguments passed to the calibration decorator.
+
+    Returns:
+        GateCalibration: Object representing the calibration definition.
+    """
+    decorator_qubit_names, decorator_angle_names = _categorize_calibration_decorator_args(
+        decorator_kwargs
+    )
+    func_args = _get_gate_args(f)
+    _validate_calibration_args(gate_function, decorator_kwargs, func_args)
```

Comment:
Would it simplify the logic if `_validate_calibration_args` took `decorator_qubit_names` and `decorator_angle_names` instead of `decorator_kwargs`?